packages/sw-cli/test/end-to-end.js

```diff
@@ -0,0 +1,105 @@
+const proxyquire = require('proxyquire');
+const path = require('path');
+const fs = require('fs');
+const vm = require('vm');
+const glob = require('glob');
+const expect = require('chai').expect;
+
+require('chai').should();
+
+describe('Test Example Projects', function() {
+  it('should be able to generate manifest for example-1', function() {
+    const exampleProject =
+      path.join(__dirname, 'example-projects', 'example-1');
+    const relativeProjPath = path.relative(process.cwd(), exampleProject);
+
+    // NOTE: No JPG
+    const fileExntensions = ['html', 'css', 'js', 'png'];
+
+    const manifestName = `${Date.now()}-manifest.js`;
+    const swName = `${Date.now()}-sw.js`;
+
+    const SWCli = proxyquire('../src/cli/index', {
+      inquirer: {
+        prompt: (questions) => {
+          switch (questions[0].name) {
+            case 'rootDir':
+              return Promise.resolve({
+                rootDir: relativeProjPath,
+              });
+            case 'cacheExtensions':
+              return Promise.resolve({
+                cacheExtensions: fileExntensions,
+              });
+            case 'fileManifestName':
+              return Promise.resolve({
+                fileManifestName: manifestName,
+              });
+            case 'serviceWorkerName':
+              return Promise.resolve({
+                serviceWorkerName: swName,
+              });
+            case 'saveConfig':
+              return Promise.resolve({
+                saveConfig: false,
+              });
+            default:
+              console.error('');
+              console.error(`Unknown question: ${questions[0].name}`);
+              console.error('');
+              return Promise.reject();
+          }
+        },
+      },
+    });
+
+    const cli = new SWCli();
+    return cli.handleCommand('generate-sw')
+    .then(() => {
+      const injectedSelf = {};
+      const manifestContent =
+        fs.readFileSync(path.join(exampleProject, manifestName));
+      vm.runInNewContext(manifestContent, {
```

Comment:
Could you provide a comment explaining why this is necessary? I'm not familiar with using the `vm` module and some more context (`NewContext`, if you will...) would be useful.